### PR TITLE
Fix e2e tests for pre sdk v0.47 chains

### PR DESCRIPTION
**Pull request by chatton**
## Description

Testing with compatibility tests: https://github.com/cosmos/ibc-go/actions/runs/3958871372

It looks like there is some issue with the memo test, but I don't believe this changes would have anything to do with those failures. I will create a follow up issue to investigate it.

follow up fix after #2672

### Commit Message / Changelog Entry

N/A

I don't think we need a message as we are fixing something that was broken by the sdk 0.47 update.

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/testconfig/testconfig.go [hunk 1]
```diff
@@ -190,42 +187,64 @@ var govGenesisFeatureReleases = semverutil.FeatureReleases{
 
 // defaultModifyGenesis will only modify governance params to ensure the voting period and minimum deposit
 // are functional for e2e testing purposes.
+// Note: this function intentionally does not use the type defined here https://github.com/tendermint/tendermint/blob/64747b2b184184ecba4f4bffc54ffbcb47cfbcb0/types/genesis.go#L39
```
Comment (damiannolan) [inline e2e/testconfig/testconfig.go:190]: should we link to a release tag instead of a commit?

File: e2e/testconfig/testconfig.go [hunk 2]
```diff
@@ -190,42 +187,64 @@ var govGenesisFeatureReleases = semverutil.FeatureReleases{
...
+// and uses a map[string]interface{} instead.
+// this is because ibctest performs the following steps when creating the genesis.json file for chains.
+// - 1. Let the chain binary create its own genesis file.
+// - 2. Apply any provided functions to modify the bytes of the file.
+// - 3. Overwrite the file with the new contents.
+// This is a problem because when the tendermint types change, marshalling into the type will cause us to lose
+// values if the types have changed in between the versis of the chain in the test and the version of tendermint
```
Comment (damiannolan) [inline e2e/testconfig/testconfig.go:197]:
```suggestion
// values if the types have changed in between the version of the chain in the test and the version of tendermint
```

**Review by damiannolan — APPROVED** [review]
LGTM! Thanks for debugging this! ❤️ 

Do you think we should add in the comment that it was is `block.TimeIotaMs` param that was being swallowed by using the `v0.37.0` type for `GenesisDoc`?

File: e2e/testconfig/testconfig.go (hunk 1 above)
Comment (chatton) [reply e2e/testconfig/testconfig.go:190]: Yeah I think that's a good idea, will update the comment!

**Review by crodriguezvega — APPROVED** [review]
Great work, @chatton!

**Pull request merged**